### ISSUE TITLE
Really bad routes getting picked

### ISSUE THREAD

**Issue by amoldeshpande** [organicmaps/organicmaps#3001]
To be fair, this is not just an OrganicMaps issue as downloading OpenStreetMaps to my Garmin GPS also picks the same (or very close) routes.

I am travelling through Central America on a motorcycle. I find that very frequently  the app will route me through a town when the highway goes around it. Other times it will put me on a dirt track, barely a road, apparently because that is a shorter distance.

Yesterday, I was travelling from Quetzaltenango, Guatemala to San Pedro de Atitlan. If you compare the routing with Google Maps you will see that OSM apps will route into the village of Santa Lucia Utatlan and south to Tzununa then along the lake shore to San Pedro. Google on the other hand will stick to the highway RD-4, on a more direct route.

Even if there is actually a way where OrganicMaps wants to go (I am skeptical), this way is obviously slower, going through villages  and probably unpaved segments.

I am mainly trying to understand why such a route gets picked and if there is any way to change the behavior. Depending on the situation, this could send me into dangerous situations and I would obviously like to minimize the chances of such a thing :) 

As I said, it is not a OrganicMaps-specific issue. My Garmin GPS and Maps.me both do the same routing here.

Thanks!

**Comment by janusbenissa** [comment]
Than you have to see in the OSM map how the roads are tagged I suppose.

**Comment by biodranik** [comment]
Yes, all algorithms use OSM tags for the input. If you find mapping errors and fix them, then routing will improve.

**Comment by amoldeshpande** [comment]
At least on the current OSM website, the bad road in my example is marked "tertiary" and the good one is marked "secondary". So I'd assume routing should work as in Google Maps ? (secondary preferred over tertiary )

Is there a way I can update the OSM map in OrganicMaps myself ? I am using on Android as well as iOS if it's possible in either.

**Comment by janusbenissa** [comment]
Don't assume thinks if you can search for it https://wiki.openstreetmap.org/wiki/Key:highway
Also look at  [smoothness](https://wiki.openstreetmap.org/wiki/Key:smoothness) and [smoothness](https://wiki.openstreetmap.org/wiki/Key:smoothness)
You can't edit the type of road in OM

**Comment by amoldeshpande** [comment]
well, the problem is that the paved/unpaved state in OrganicMaps does not reflect the current state of the roads. I cannot find anything in ID editor other than "surface" for these roads.  Is there a way to look at the OrganicMaps' version online ?

Maybe a better question is what tags/properties are used by the routing algorithm to pick a route ? I was literally taken into a cornfield on a dirt track because it was a shortcut to a highway.  

A link to RTFM about how these routing algorithms pick roads would be fine. 

thanks.

**Comment by janusbenissa** [comment]
> well, the problem is that the paved/unpaved state in OrganicMaps does not reflect the current state of the roads.
Again, you have to edit that in OSM

**Comment by amoldeshpande** [comment]
I don't think you understand what I am asking. 

Since the current state of roads in OSM (website) does not match OrganicMaps it is not possible for me to try and figure out why OrganicMaps is routing differently from Google, just by looking at OSM tags and properties.

So, either I need to look at the maps version that OrganicMaps is using, or look into how the algorithm works so I can at least have a theory. 

I am not trying to change anything. I want to "diff" the maps in that area to see if I can spot something.

**Comment by biodranik** [comment]
You can check the date of the OSM snapshot in OM by pressing "?" button.
If you see an obvious error in OM and see that the online version in OSM is correct, and wasn't touched since the snapshot date, then it could be an OM bug. Please tell us the starting and ending points of the suspected wrong route.

**Comment by amoldeshpande** [comment]
thanks. I spent some more time diving into the the details of the segments of the route and it looks like there are certain short parts that OSM has outdated surface information on (dirt vs paved). 

That could be triggering the different routing, even though I don't have "avoid dirt roads" checked in options. 

I have indicated the start and end points in my first post, and where the route differs from Google Maps. Any insight into why the routing algorithm is picking the wrong way would be appreciated.

**Comment by vng** [comment]
surface=dirt adds _big_ penalty for roads:
https://www.openstreetmap.org/way/353732635
https://www.openstreetmap.org/way/548777203
https://www.openstreetmap.org/way/548777205

On the other hand, this way https://www.openstreetmap.org/way/370746655 has surface=unpaved.

Now the app treats _dirt_ worse than _unpaved_. Can't say for sure is it ok or not ..

**Comment by amoldeshpande** [comment]
thanks for the explanation. I would suspect people tag unpaved and dirt interchangeably, but that's a hard one to be certain of.

It would be nice to have a "motorcycle" mode where if "avoid dirt roads" is unchecked, it adds no penalty for dirt/unpaved.

**Comment by endim8** [comment]
hmm, unpaved is effectively a generic legacy tag. 

These are the more specific values of the legacy tags:

[Paved](https://wiki.openstreetmap.org/wiki/Tag:surface%3Dpaved) (legacy):
* asphalt
* fine_gravel
* chipseal
* concrete
* paving_stones
* etc...

[Unpaved ](https://wiki.openstreetmap.org/wiki/Tag:surface%3Dunpaved)(legacy):
* mud
* dirt
* gravel
* grass
* compacted
* etc...

imo, unpaved should be considered roughly equal to surface=dirt 👍
